Google.PowerShell.IntegrationTests/Container/Container.GkeCluster.Tests.ps1

```diff
@@ -162,10 +184,151 @@ Describe "New-GkeNodeConfig" {
+
+    # Cluster Two Creation.
+    $script:networkName = "test-network-$r"
+
+    # Create a network and extract out subnet that corresponds to region "us-central1".
+    gcloud compute networks create $networkName 2>$null
+    $region = "us-central1"
+    $network = Get-GceNetwork $networkName
+    $subnet = $network.Subnetworks | Where-Object {$_.Contains($region)}
+    $subnet -match "subnetworks/([^/]*)"
+    $script:subnetName = $Matches[1]
+
+    $clusterTwoScriptBlock = {
+        param($cmdletPath, $clusterName, $clusterNetwork, $clusterSubnetwork, $clusterAdditionalZone)
+        . $cmdletPath;
+        Install-GCloudCmdlets | Out-Null;
+        Add-GkeCluster -ClusterName $clusterName -MachineType n1-standard-1 `
+                       -Network $clusterNetwork -Subnetwork $clusterSubnetwork `
+                       -InitialNodeCount 4 -AdditionalZone $clusterAdditionalZone
+    }
+
+    $script:clusterTwoName = "gcp-new-gkecluster-2-$r"
+    $clusterTwoAdditionalZone = "us-central1-c"
+    $clusterTwoJob = Start-Job -ScriptBlock $clusterTwoScriptBlock `
+                               -ArgumentList @($gcloudCmdletsPath, $clusterTwoName, $network.Name,
+                                               $subnetName, $clusterTwoAdditionalZone)
+
+    # Cluster Three Creation.
+    $clusterThreeScriptBlock = {
+        param($cmdletPath, $clusterName, $clusterZone, $clusterUsername, $clusterPassword)
+        . $cmdletPath;
+        Install-GCloudCmdlets | Out-Null;
+        $password = ConvertTo-SecureString $clusterPassword -AsPlainText -Force
+        $credential = New-Object System.Management.Automation.PSCredential ($clusterUsername, $password)
+        Add-GkeCluster -ClusterName $clusterName `
+                       -MasterCredential $credential `
+                       -Zone $clusterZone
+    }
+
+    $script:clusterThreeName = "gcp-new-gkecluster-3-$r"
+    $clusterThreeZone = "europe-west1-b"
+    $clusterThreeUsername = Get-Random
+    $clusterThreePassword = Get-Random
+    $clusterThreeJob = Start-Job -ScriptBlock $clusterThreeScriptBlock `
+                                 -ArgumentList @($gcloudCmdletsPath, $clusterThreeName,
+                                                 $clusterThreeZone, $clusterThreeUsername,
+                                                 $clusterThreePassword)
+
+    # For cluster with node config, the script block is having trouble converting $nodeConfig object
+    # so we will just create it here (but the other clusters are being created in the background so that's ok.
+    $clusterFourName = "gcp-new-gkecluster-4-$r"
+    $clusterFourZone = "us-west1-a"
+    $clusterFourConfig = New-GkeNodeConfig -DiskSizeGb 20 `
+                                            -LocalSsdCount 3 `
+                                            -Label @{"Release" = "stable"} `
+                                            -MachineType n1-highcpu-4
+
+    Add-GkeCluster -ClusterName $clusterFourName `
```

Comment:
There's a lot going on here.  Can some of this be factored out into helper functions?